docs/src/test/scala/docs/http/scaladsl/server/directives/StyleGuideExamplesSpec.scala

```diff
@@ -0,0 +1,192 @@
+/*
+ * Copyright (C) 2009-2020 Lightbend Inc. <https://www.lightbend.com>
+ */
+
+package docs.http.scaladsl.server.directives
+
+import akka.http.scaladsl.server._
+import docs.CompileOnlySpec
+
+class StyleGuideExamplesSpec extends RoutingSpec with CompileOnlySpec {
+
+  "path" should {
+
+    "be outermost" in {
+      // #path-outermost
+      // prefer
+      val prefer = path("item" / "listing") & get
+      // over
+      val over = get & path("item" / "listing")
+      // #path-outermost
+    }
+
+    "use prefix" in {
+      // #path-prefix
+      // prefer
+      val prefer =
+        pathPrefix("item") {
+          concat(
+            path("listing") {
+              get {
+                complete("")
+              }
+            },
+            path("show" / Segment) { itemId =>
+              get {
+                complete("")
+              }
+            }
+          )
+        }
+      // over
+      val over: Route =
+        concat(
+          path("item" / "listing") {
+            get {
+              complete("")
+            }
+          },
+          path("item" / "show" / Segment) { itemId =>
+            get {
+              complete("")
+            }
+          }
+        )
+      // #path-prefix
+    }
+
+    "split" in {
+      // #path-compose
+      // prefer
+      // 1. First, create partial matchers (with a relative path)
+      val itemRoutes: Route =
+        concat(
+          path("listing") {
+            get {
+              complete("")
+            }
+          },
+          path("show" / Segment) { itemId =>
+            get {
+              complete("")
+            }
+          }
+        )
+
+      val customerRoutes: Route =
+        concat(
+          path("customer" / IntNumber) { customerId =>
```

Comment:
This should have been
```scala
path(IntNumber) { customerId =>
```
to match the Java and avoid a confusing `/customer/customer/123` route.

Reply:
Thank you. By coincidence we got a fix for this the other day in https://github.com/akka/akka-http/pull/3448

Reply:
Perfect! Thanks.